calcom/cal.com
# Self-hosted onboarding first admin wizard

## What does this PR do?

Implements a wizard to create the first user (admin) as an on-boarding step to the app. The on-boarding will continue to expand in the near future which is why it was implemented this way.

UI:
<kbd><img width="937" alt="image" src="https://user-images.githubusercontent.com/467258/180276506-4ee642e3-fea0-40c7-b108-32378a906269.png"></kbd>

Can't be used again once at least one user exists in the DB:
<kbd><img width="937" alt="image" src="https://user-images.githubusercontent.com/467258/180276790-6175714c-b624-454c-a337-466956dcd75b.png"></kbd>

Implements #3384 



**Environment**: Staging(main branch) / Production

## Type of change


## How should this be tested?

Remove all users from the DB and go to the login page (not having a session active). It will show the new onboarding wizard. You can also test going to `/auth/setup` having users in the DB to see the second screen above, indicating the step is done (not needed). The API that supports this new functionality also checks if there is no user in the DB to be able to enable creating an admin user through the new onboarding wizard.
